### ISSUE TITLE
DarkMode cant disable

### ISSUE THREAD

**Issue by Anagastes** [roundcube/roundcubemail#8186]
Hey, my template does not support dark mode and should not in the future.

But with a config entry I can't disable the button for users.

```
$config['dark_mode_support'] = false;
$config['dark_mode_support'] = 0;
```

Neither works.

I had to manually remove the button here now.

skins/XXX/templates/includes/menu.html

```
<!-- <roundcube:if condition="config:dark_mode_support" />
                                <roundcube:add_label name="darkmode" />
                                <roundcube:add_label name="lightmode" />
                                <roundcube:button name="theme" label="darkmode" type="link" innerClass="inner" class="theme dark" />
                        <roundcube:endif />-->
```

Please enable a simple disable of the darkmode. :)

**Comment by alecpl** [comment]
You can set it via the skin `meta.json` file. Indeed, it's not possible to change this via the config.inc.php file, because skin setup options have a higher prio.

**Comment by alecpl** [comment]
If you have your custom skin `meta.json` is the way to go. Isn't that working?

**Comment by Anagastes** [comment]
> If you have your custom skin `meta.json` is the way to go. Isn't that working?

Ah ok, but "dark_mode_support false" didnt work for me

The json now looks like this for me. The json now looks like this for me. Base is elastic (only minimally customized, css color etc.).

```
{
        "name": "Custom",
        "author": "Aleksander Machniak modified by Anagastes M. W.",
        "license": "Creative Commons Attribution-ShareAlike",
        "license-url": "http://creativecommons.org/licenses/by-sa/3.0/",
        "config": {
                "supported_layouts": ["widescreen"],
                "jquery_ui_colors_theme": "bootstrap",
                "embed_css_location": "/styles/embed.css",
                "editor_css_location": "/styles/embed.css",
                "dark_mode_support": false,
                "media_browser_css_location": "none",
                "additional_logo_types": ["dark", "small", "small-dark"]
        },
        "meta": {
                "viewport": "width=device-width, initial-scale=1.0, shrink-to-fit=no, maximum-scale=1.0",
                "theme-color": "#f4f4f4",
                "msapplication-navbutton-color": "#f4f4f4"
        }
}
```

**Comment by alecpl** [comment]
I just verified, setting `config.dark_mode_support: false` in `meta.json` of the skin (that extends Elastic) works.

**Comment by Anagastes** [comment]
> I just verified, setting `config.dark_mode_support: false` in `meta.json` of the skin (that extends Elastic) works.

Okay, not for me. Had changed from 1.4.11 to 1.5 rc.

Here you can see how it looks although false is set.

https://mail.miluriel.info

**Comment by Anagastes** [comment]
ok, i see in chrome it works but not in firefox. There the dark mode is loaded by default.

**Comment by Anagastes** [comment]
> I just verified, setting `config.dark_mode_support: false` in `meta.json` of the skin (that extends Elastic) works.

I think I found a possible trigger.

https://www.reddit.com/r/firefox/comments/dgke1s/help_is_there_a_way_to_disable_dark_mode_websites/

But disabling it did not help in Firefox. Apparently Firefox recognizes that there is a darkmode and tries to load it.

btw. KDE in dark is in use here

**Comment by alecpl** [comment]
The option hides the menu button, but does not disable the functionality which is using a cookie. I suppose this could be improved.

**Comment by Anagastes** [comment]
> The option hides the menu button, but does not disable the functionality which is using a cookie. I suppose this could be improved.

Ok, thanks for opening the request :)

I guess this won't be added until the stable release.

Is there a way to remove the darkmode from elastic until then? So that skins based on it can turn it off temporarily?

**Comment by alecpl** [comment]
Actually there is: `@dark-mode-enabled: false;` in `styles/_variables.less`.

**Comment by Anagastes** [comment]
Hey, I found another bug based on this (in a completely fresh installation).

I have adjusted the meta.json as well as the variable.less.

@dark-mode-enabled: false;
"dark_mode_support: false,

and regenerated the whole thing directly with this.

https://lesscss.org/#

However, a black background appears in the mail overview and Firefox when the desktop is in dark mode.

![black bg](https://imgshare.io/images/2021/09/01/ksnip_20210902-011003.png "black bg")

**Comment by Anagastes** [comment]
Unfortunately I am not a php developer. I have therefore made the simplest workaround for me.

In the file watermark.html

```html
html.dark-mode > body {
  background-color: #21292c;
  background-blend-mode: soft-light;
}
```

Changed to.

```html
html.dark-mode > body {
  /*background-color: #21292c;*/
  background-blend-mode: soft-light;
  background: url(images/logo.svg) center no-repeat rgba(255, 255, 255, .85);
  background-size: 30%;
}
```

I'll keep testing and see if any darkmode elements pop up anywhere yet :)

**Comment by Anagastes** [comment]
Big sorry that I just dig up so much :D

But in mobile view the menu-header seems to be loaded in dark mode.

@color-popover-mobile-header-background:    @color-main-dark;

So any light color set under @color-main will be darkened.

```css
@color-main:                #XXXX;
@color-main-dark: darken(@color-main, 35%);
```

So no matter if light or dark template, it doesn't match guaranteed in the light area at the latest.

I suspect problems here if you are forced to maintain both a dark and a light template.

Therefore, the area in the light mode should not be darkened. :)

As an example. If I set it like this.

```css
@color-main:                #ff5d19;
@color-main-dark: darken(@color-main, 35%);
```

Then it looks great in dekstop mode.

But mobile it looks like this.

![mobile issue](https://imgshare.io/images/2021/09/01/ksnip_20210902-023836.png "mobile issue")

If I now set the part to the identical value of the main color, it fits.

```css
@color-popover-mobile-header-background:    @color-layout-header-background;
```

![mobile issue](https://imgshare.io/images/2021/09/01/ksnip_20210902-024124.png "mobile issue fix")
